File: src/main/java/com/goudourasv/http/lectures/dto/LectureCreate.java
package com.goudourasv.http.lectures.dto;

import org.hibernate.validator.constraints.URL;

import javax.validation.constraints.NotBlank;
import java.net.URI;
import java.time.Instant;
import java.util.UUID;

public class LectureCreate {
    @NotBlank
    private String title;
//    @URL
    private URI uri;
    private Instant startTime;
    private Instant endTime;
    private UUID courseId;

    public LectureCreate(String title, URI uri, Instant startTime, Instant endTime, UUID courseId) {
        this.title = title;
        this.uri = uri;
        this.startTime = startTime;
        this.endTime = endTime;
        this.courseId = courseId;
    }

    public String getTitle() {
        return title;
    }

    public Instant getStartTime() {
        return startTime;
    }

    public Instant getEndTime() {
        return endTime;
    }

    public UUID getCourseId() {
        return courseId;
    }

    public URI getUri() {
        return uri;
    }
}

File: src/main/java/com/goudourasv/http/lectures/dto/LectureCreate.kt
package com.goudourasv.http.lectures.dto

import java.net.URI
import java.time.Instant
import java.util.*
import javax.validation.constraints.NotBlank

data class LectureCreate(
    @NotBlank val title: String,
    val uri: URI?,
    val startTime: Instant,
    val endTime: Instant,
    val courseId: UUID,
) {
}